moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1415,12 +1418,24 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
                          context.trajectory_parts_[longest_part].multi_dof_joint_trajectory.points.size()))
           longest_part = i;
       }
-      expected_trajectory_duration = std::max(d, expected_trajectory_duration);
-    }
-    // add 10% + 0.5s to the expected duration; this is just to allow things to finish propery
 
-    expected_trajectory_duration = expected_trajectory_duration * allowed_execution_duration_scaling_ +
-                                   ros::Duration(allowed_goal_duration_margin_);
+      // use controller-specific values if defined
```

Comment:
Please add a comment here, that the global parameters are dynamically reconfigurable (and thus might change from call to call). If there are controller-specific values defined, they override any global parameters - even if set through dynamic reconfigure.

That's the point, I'm feeling still uncomfortable with. If the user has changed the global parameters, they will not take effect and the user isn't notified about that fact...
Did you considered to use the minimum of global and specific?

Reply:
@rhaschke We were discussing over here where exactly you wanted to have more documentation and @2scholz added more explanation to the tutorial already: https://github.com/ros-planning/moveit_tutorials/pull/161 .

I see your concerns about dynamic_reconfigure not working, but the only way to get around this is to patch dynamic_reconfigure and allow runtime variables in there. I don't like that very much.

Neither minimum nor maximum operator make sense in practice.

Reply:
I think, there should be a reminder in the source code (at indicated line 1422), that controller-specific values will override dynamically configurable globals. I don't yet have a solution for this as well, but there should be a reminder in the source code if we come up with a solution later...

Eventually there should also be a warning (once) if global parameters have changed through dynamic reconfigure, but are overridden by controller-specific ones. All you need to do for this, is to keep track of the dyn-reconfig-callbacks (I think there is always one to set the initial value).

Reply:
I just added a comment there.